lib/color_spinor_field.cpp

```diff
@@ -27,20 +27,40 @@ namespace quda {
     dslash_constant(static_cast<DslashConstant *>(safe_malloc(sizeof(DslashConstant)))),
     bytes(0),
     norm_bytes(0),
+    location(param.location),
     even(0),
     odd(0),
     composite_descr(param.is_composite, param.composite_dim, param.is_component, param.component_id),
     components(0)
   {
+    if (param.location == QUDA_INVALID_FIELD_LOCATION) errorQuda("Location not set");
```

Comment:
might  be worth to do a full clang-format on this file given how much has changed

Reply:
Will do that when I apply clang-format to the PR.  That'll make @weinbe2 happy 😉 .

Reply:
4a4dab7df8d4b1fede2f7a50b314181c8433e4c5